freqtrade/freqai/data_kitchen.py

```diff
@@ -1027,3 +1002,78 @@ def get_unique_classes_from_labels(self, dataframe: DataFrame) -> None:
         if self.unique_classes:
             for label in self.unique_classes:
                 self.unique_class_list += list(self.unique_classes[label])
+
+# Methods called by interface.py (load_freqai_model())
+
+
+def download_all_data_for_training(timerange: TimeRange,
+                                   dp: DataProvider, config: dict) -> None:
+    """
+    Called only once upon start of bot to download the necessary data for
+    populating indicators and training the model.
+    :param timerange: TimeRange = The full data timerange for populating the indicators
+                                    and training the model.
+    :param dp: DataProvider instance attached to the strategy
+    """
+    all_pairs = copy.deepcopy(
+        config["freqai"]["feature_parameters"].get("include_corr_pairlist", [])
+    )
+    for pair in config.get("exchange", "").get("pair_whitelist"):
+        if pair not in all_pairs:
+            all_pairs.append(pair)
```

Comment:
i think this can be reduced to a call to `dynamic_expand_pairlist()` - which odes exactly this logic for `download-data`.

Reply:
Fixed.